docs/import/color_function_util.js

```diff
@@ -1,8 +1,165 @@
+let globalTd;
+
+/**
+ * Fills out the #single #continuous and #discrete divs with the relevant
+ * DOM elements with attached on-change handlers.
+ */
+function populateColorFunctions() {
+  const colorFunctionDiv = $('#color-fxn-editor');
+  colorFunctionDiv.prepend(createRadioFor(ColorStyle.SINGLE));
+  colorFunctionDiv.prepend(createRadioFor(ColorStyle.DISCRETE));
+  colorFunctionDiv.prepend(createRadioFor(ColorStyle.CONTINUOUS));
+
+  const singleColorPicker = createColorPicker('single-color-picker');
+  singleColorPicker.on('change', () => setColor(singleColorPicker));
+  $('#single').append(
+      createLabelFor(singleColorPicker, 'color'), singleColorPicker);
+
+  const continuousColorPicker = createColorPicker('continuous-color-picker');
+  continuousColorPicker.on('change', () => setColor(continuousColorPicker));
+  const continuousPropertyPicker =
+      $(document.createElement('select'))
+          .prop('id', 'continuous-property-picker');
+  continuousPropertyPicker.on(
+      'change', () => setProperty(continuousPropertyPicker));
+  $('#continuous')
+      .append(
+          createLabelFor(continuousColorPicker, 'base color'),
+          continuousColorPicker, $(document.createElement('br')),
+          createLabelFor(continuousPropertyPicker, 'property'),
+          continuousPropertyPicker);
+
+  const discretePropertyPicker = $(document.createElement('select'))
+                                     .prop('id', 'discrete-property-picker');
+  discretePropertyPicker.on('change', () => {
+    setProperty(discretePropertyPicker);
+    populateDiscreteColorPickers();
+  });
+  const discreteColorPickers =
+      $(document.createElement('ul')).prop('id', 'discrete-color-pickers');
+  $('#discrete')
+      .append(
+          createLabelFor(discretePropertyPicker, 'property'),
+          discretePropertyPicker, discreteColorPickers);
+}
+
+/**
+ * Updates the 'field' property which is shared by the continuous and discrete
+ * color schemas.
+ * @param {JQuery<HTMLElement>} picker
+ */
+function setProperty(picker) {
+  getColorFunction()['field'] = picker.val();
+  updateLayersInFirestore();
+}
+
+/**
+ * Updates an individual color choice for a single value in the discrete schema.
+ * @param {JQuery<HTMLElement>} picker
+ */
+function setDiscreteColor(picker) {
+  const colorFunction = getColorFunction();
+  const propertyValue = picker.data(discreteColorPickerDataKey);
+  if (!colorFunction['colors']) {
+    colorFunction['colors'] = {};
+  }
+  colorFunction['colors'][propertyValue] = picker.val();
+  updateLayersInFirestore();
+  populateColorTd();
+}
+
+/**
+ * Updates the 'color' property which is shared by the continuous and single
+ * color schemas.
+ * @param {JQuery<HTMLElement>} picker
+ */
+function setColor(picker) {
+  getColorFunction()['color'] = picker.val();
+  updateLayersInFirestore();
+  populateColorTd();
+}
+
+/**
+ * Refreshes the current color cell.
+ */
+function populateColorTd() {
```

Comment:
As noted offline, it seems like this should be called on every single "pick", including changing the relevant property (in setProperty)? I wonder if the Firestore update should just be wrapped in a function that also updates the td. That way they'll never get out of sync.

Reply:
Done.